src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java

```diff
@@ -64,16 +67,23 @@ public Session apply(CreateSessionDTO dto) throws AuthenticationException {
             Thing thing = thingRegistry.getOrCreateThing(thingName);
 
             VerifyThingAttachedToCertificate verify = useCases.get(VerifyThingAttachedToCertificate.class);
-            Boolean thingAttachedResult = verify.apply(
+            VerifyThingAttachedToCertificate.Result result = verify.apply(
                     new VerifyThingAttachedToCertificateDTO(thingName, certificate.get().getCertificateId()));
 
-            if (thingAttachedResult) {
+            logger.atDebug()
+                    .kv("thingIsAttached", result.isThingAttachedToCertificate())
+                    .kv("lastAttachedOn", result.getLastAttached())
+                    .kv("attachmentExpiration", result.getAttachmentExpiration())
+                    .kv("source", result.getVerificationSource())
+                    .log("Attachment verification result");
+
+            if (result.isThingAttachedToCertificate()) {
```

Comment:
check for expiration here too, not just that it is attached?

Reply:
ah yeah that was a miss, fixing

Reply:
spoke too soon, it's covered here: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/238/files#diff-d564d0cc489527df3742905afa00509b6cfb6ebdb07597c35138150a4b2c1a45R56

probably needs a wording

Reply:
changed wording from `thingAttachedToCertificate` to `thingHasValidAttachmentToCertificate`,  hopefully more clear